connectors/sources/sharepoint.py

```diff
@@ -0,0 +1,715 @@
+            file_name(string, optional): Name of file. Defaults to None.
+
+        Returns:
+            dictionary: Modified document with the help of adapter schema.
+        """
+        document = {"type": document_type}
+
+        if document_type in [LISTS, DOCUMENT_LIBRARY]:
+            document["url"] = urljoin(
+                self.host_url, item["RootFolder"]["ServerRelativeUrl"]
+            )
+
+        elif document_type == DRIVE_ITEM:
+            document.update(
+                {
+                    "_id": item["GUID"],
+                    "size": item.get("File", {}).get("Length"),
+                    "url": urljoin(self.host_url, item[item_type]["ServerRelativeUrl"]),
+                    "type": item_type,
+                }
+            )
+
+        elif document_type == LIST_ITEM:
+            document.update(
+                {
+                    "_id": item["_id"] if "_id" in item.keys() else item["GUID"],
+                    "file_name": file_name,
+                    "size": item.get("size"),
+                    "url": item["url"],
+                }
+            )
+
+        for elasticsearch_field, sharepoint_field in SCHEMA[document_type].items():
+            document[elasticsearch_field] = (
+                item[item_type][sharepoint_field]
+                if document_type in [DRIVE_ITEM]
+                else item[sharepoint_field]
+            )
+
+        return document
+
+    async def ping(self):
+        """Verify the connection with Sharepoint"""
+
+        self._validate_configuration()
+
+        if not self.ssl_disabled:
+            self.ssl_ctx = self._ssl_context(certificate=self.certificate)
+
+        self.session = await self._generate_session()
+        try:
+            await anext(
+                self._api_call(
+                    url_name=PING,
+                    site_collections=self.site_collections[0],
+                    host_url=self.host_url,
+                )
+            )
+            logger.info("Successfully connected to the Sharepoint")
+        except Exception:
+            logger.exception("Error while connecting to the Sharepoint")
+            raise
+
+    async def get_content(
+        self, document, file_relative_url, site_url, timestamp=None, doit=None
+    ):
+        """Get content of list items and drive items
+
+        Args:
+            document (dictionary): Modified document.
+            file_relative_url (str): Relative url of file
+            site_url (str): Site path of sharepoint
+            timestamp (timestamp, optional): Timestamp of item last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to None.
+
+        Returns:
+            dictionary: Content document with id, timestamp & text.
+        """
+
+        if not (self.enable_content_extraction and doit and document["size"]):
```

Comment:
cc @tarekziade - is `self.enable_content_extraction` a valid use-case?

I know that we have deployment-wide settings in pipeline to enable/disable binary content extraction. Why have additional configurable field for it?

Reply:
@tarekziade ping